archlinuxcn/3dslicer-bin/lilac.py

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python3
+
+from lilaclib import *
+import subprocess
+
+def pre_build():
+  update_pkgver_and_pkgrel(_G.newver.lstrip('v'))
+  cmd = "curl https://download.slicer.org/ |grep bitstream |cut -d '/' -f 3 |cut -d '\"' -f 1 |sed -n 3p"
+  newstr = subprocess.run(cmd, stdout=subprocess.PIPE, shell=True).stdout.decode('utf-8')[0:-1]
+
+  cmd = "grep bitstream PKGBUILD|cut -f 5 -d '/'
+  oldstr = subprocess.run(cmd, stdout=subprocess.PIPE, shell=True).stdout.decode('utf-8')[0:-2]
+
+  if oldstr == newstr:
+    raise RuntimeError("URL is not updated yet.")
```

Comment:
You can use a `prepare` function to check whether a build should be done, and return a string when it's not ready (example in rust-nightly). You don't receive error reports for skipped builds.

Also, you can use Python to retrieve the URL and update PKGBUILD. There is a requests.Session object named `s` available to send requests. There is a `edit_file` function for you to update PKGBUILD easily.